up-linux-streamer-plugin/src/lib.rs

```diff
@@ -0,0 +1,279 @@
+#![recursion_limit = "256"]
+
+use async_std::task;
+use futures::select;
+use std::collections::HashMap;
+use std::convert::TryFrom;
+use std::env;
+use std::fs::canonicalize;
+use std::ops::Deref;
+use std::path::PathBuf;
+use std::sync::{
+    atomic::{AtomicBool, Ordering::Relaxed},
+    Arc, Mutex,
+};
+use std::time::Duration;
+use tracing::{debug, info};
+use tracing::{error, trace};
+use up_rust::UTransport;
+use up_streamer::{Endpoint, UStreamer};
+use up_transport_vsomeip::UPTransportVsomeip;
+use up_transport_zenoh::UPClientZenoh;
+use zenoh::plugins::{RunningPluginTrait, ZenohPlugin};
+use zenoh::prelude::r#async::*;
+use zenoh::runtime::Runtime;
+use zenoh_core::zlock;
+use zenoh_plugin_trait::{plugin_long_version, plugin_version, Plugin, PluginControl};
+use zenoh_result::{bail, ZResult};
+
+// The struct implementing the ZenohPlugin and ZenohPlugin traits
+pub struct ExamplePlugin {}
+
+// declaration of the plugin's VTable for zenohd to find the plugin's functions to be called
+#[cfg(feature = "dynamic_plugin")]
+zenoh_plugin_trait::declare_plugin!(ExamplePlugin);
+
+// A default selector for this example of storage plugin (in case the config doesn't set it)
+// This plugin will subscribe to this selector and declare a queryable with this selector
+const DEFAULT_SELECTOR: &str = "demo/example/**";
+
+impl ZenohPlugin for ExamplePlugin {}
+impl Plugin for ExamplePlugin {
+    type StartArgs = Runtime;
+    type Instance = zenoh::plugins::RunningPlugin;
+
+    // A mandatory const to define, in case of the plugin is built as a standalone executable
+    const DEFAULT_NAME: &'static str = "up_linux_streamer";
+    const PLUGIN_VERSION: &'static str = plugin_version!();
+    const PLUGIN_LONG_VERSION: &'static str = plugin_long_version!();
+
+    // The first operation called by zenohd on the plugin
+    fn start(name: &str, runtime: &Self::StartArgs) -> ZResult<Self::Instance> {
+        zenoh_util::try_init_log_from_env();
+        trace!("up-linux-streamer-plugin: start");
+        let config = runtime.config().clone();
+        let config_keys = config.keys();
+        trace!("name: {name}");
+        trace!("config_keys: {config_keys:?}");
+        let plugins = config.get("plugins");
+        match plugins {
+            Ok(config_guard) => {
+                // @CY / @Luca - Looks like this is None for some reason?
+                trace!("config_guard: {:?}", config_guard.downcast_ref::<Config>());
+            }
+            Err(err) => {
+                error!("Unable to get plugins: {err:?}");
+            }
+        }
+        trace!("Made past match on plugins");
+
+        // @CY / @Luca -- seems to crash here?
+        // let config_plugin = runtime.config().get_json("plugins");
```

Comment:
Adding a few spots to tag in @evshary & @mallets for questions about reading the Zenoh Config from the Runtime.